File: src/main/cplusplus/net/coderextreme/data/DiamondManLOA0_patched2.cpp
/* delete next 5 lines for Visual Studio */
#define FALSE false
#define TRUE true
#define WINAPI
#define AFX_EXT_CLASS
#define EXPORT32
#include "C:/x3d-code/www.web3d.org/x3d/stylesheets/cpp/concretes.h"
int main(int argc, char ** argv) {
CX3DScene m_pScene;
CBrowser browser = X3D.getBrowser();
CX3D* X3D0 = new CX3D();
CGroup* group = (CGroup*)(m_pScene.createNode("Group"));
group->addChildren(X3D0);
X3D0->setProfile("Immersive");
X3D0->setVersion("4.0");
Chead* head1 = new Chead();
Cmeta* meta2 = new Cmeta();
meta2->setName("title");
meta2->setContent("DiamondManLOA0.x3d");
head1->addMeta(*meta2);

Cmeta* meta3 = new Cmeta();
meta3->setName("description");
meta3->setContent("HAnim skeletal structure for Level of Action (LOA) zero, with one diamond at the base node for the structure. HumanoidRoot only, so this is the minimum legal HAnim humanoid.");
head1->addMeta(*meta3);

Cmeta* meta4 = new Cmeta();
meta4->setName("creator");
meta4->setContent("Matthew T. Beitler");
head1->addMeta(*meta4);

Cmeta* meta5 = new Cmeta();
meta5->setName("translator");
meta5->setContent("Joel S. Pawloski");
head1->addMeta(*meta5);

Cmeta* meta6 = new Cmeta();
meta6->setName("created");
meta6->setContent("12 November 2001");
head1->addMeta(*meta6);

Cmeta* meta7 = new Cmeta();
meta7->setName("modified");
meta7->setContent("Thu, 14 Dec 2023 22:20:20 GMT");
head1->addMeta(*meta7);

Cmeta* meta8 = new Cmeta();
meta8->setName("motto");
meta8->setContent("(a) \"Diamonds are a girl's best friend.\" (b) \"Gosh, it sure is chilly in here.\"");
head1->addMeta(*meta8);

Cmeta* meta9 = new Cmeta();
meta9->setName("reference");
meta9->setContent("HAnim 2.0 specification, Appendix A: Nominal human body dimensions and levels of articulation (LOAs)");
head1->addMeta(*meta9);

Cmeta* meta10 = new Cmeta();
meta10->setName("reference");
meta10->setContent("https://www.web3d.org/documents/specifications/19774-1/V2.0/HAnim/BodyDimensionsAndLOAs.html");
head1->addMeta(*meta10);

Cmeta* meta11 = new Cmeta();
meta11->setName("reference");
meta11->setContent("https://www.web3d.org/documents/specifications/19774-1/V2.0/HAnim/BodyDimensionsAndLOAs.html#LevelOfArticulationZero");
head1->addMeta(*meta11);

Cmeta* meta12 = new Cmeta();
meta12->setName("reference");
meta12->setContent("HAnim 1.1 specification, Appendix A: Suggested Body Dimensions and Levels of Articulation, Level of Articulation Zero");
head1->addMeta(*meta12);

Cmeta* meta13 = new Cmeta();
meta13->setName("reference");
meta13->setContent("http://HAnim.org/Specifications/HAnim1.1/appendices.html#appendixa");
head1->addMeta(*meta13);

Cmeta* meta14 = new Cmeta();
meta14->setName("reference");
meta14->setContent("http://HAnim.org/Specifications/HAnim1.1/JointCenters1_1_LOA0.wrl");
head1->addMeta(*meta14);

Cmeta* meta15 = new Cmeta();
meta15->setName("reference");
meta15->setContent("http://HAnim.org/Specifications/HAnim1.1/JointCenters1_1_LOA0-diamond.wrl");
head1->addMeta(*meta15);

Cmeta* meta16 = new Cmeta();
meta16->setName("reference");
meta16->setContent("http://ece.uwaterloo.ca/~HAnim");
head1->addMeta(*meta16);

Cmeta* meta17 = new Cmeta();
meta17->setName("reference");
meta17->setContent("http://www.cis.upenn.edu/~badler/anthro/89-71.pdf");
head1->addMeta(*meta17);

Cmeta* meta18 = new Cmeta();
meta18->setName("reference");
meta18->setContent("http://www.cis.upenn.edu/~badler/anthro/89-71.ps");
head1->addMeta(*meta18);

Cmeta* meta19 = new Cmeta();
meta19->setName("reference");
meta19->setContent("http://www.cis.upenn.edu/~beitler");
head1->addMeta(*meta19);

Cmeta* meta20 = new Cmeta();
meta20->setName("Image");
meta20->setContent("humanoid_landmark_locations.gif");
head1->addMeta(*meta20);

Cmeta* meta21 = new Cmeta();
meta21->setName("Image");
meta21->setContent("http://HAnim.org/Specifications/HAnim1.1/humanoid_landmark_locations.gif");
head1->addMeta(*meta21);

Cmeta* meta22 = new Cmeta();
meta22->setName("identifier");
meta22->setContent("https://www.web3d.org/x3d/content/examples/HumanoidAnimation/Templates/DiamondManLOA0.x3d");
head1->addMeta(*meta22);

Ccomponent* component23 = new Ccomponent();
component23->setName("HAnim");
component23->setLevel(1);
head1->addComponent(*component23);

X3D0->setHead(*head1);

CScene* Scene24 = new CScene();
CWorldInfo* WorldInfo25 = (CWorldInfo *)(m_pScene.createNode("WorldInfo"));
WorldInfo25->setTitle("HANIM 2.0 Default Joint Centers, LOA0");
WorldInfo25->setInfo(new CString[1]{"HAnim 2.0 Default Joint Centers, Level Of Articulation (LOA) 0 -------------------------------------------------------- HANIM 1.1 (VRML 2.0) Author name: eMpTy (a.k.a. Matthew T. Beitler) HANIM 1.1 (VRML 2.0) Author email: beitler@graphics.cis.upenn.edu or beitler@acm.org HANIM 1.1 (VRML 2.0) Author homepage: http://www.cis.upenn.edu/~beitler HANIM 1.1 (VRML 2.0) Compliance Date: May 12, 1999 HANIM 1.1 Compliance Information: http://ece.uwaterloo.ca/~HAnim/ Construction Info (joint centers): The joint centers of this figure are based on the work of Norman Badler, director of the Center for Human Modeling and Simulation at the University of Pennsylvania. The original document which these joint centers are based on can be found at: http://www.cis.upenn.edu/~badler/anthro/89-71.ps, .pdf"}, 1);
group->addChildren(*WorldInfo25);

CNavigationInfo* NavigationInfo26 = (CNavigationInfo *)(m_pScene.createNode("NavigationInfo"));
NavigationInfo26->setSpeed(1.5);
group->addChildren(*NavigationInfo26);

CViewpoint* Viewpoint27 = (CViewpoint *)(m_pScene.createNode("Viewpoint"));
Viewpoint27->setDescription("Diamond Man, LOA 0");
Viewpoint27->setPosition(new float[3]{0,1,3});
Viewpoint27->setCenterOfRotation(new float[3]{0,1,0});
group->addChildren(*Viewpoint27);

CHAnimHumanoid* HAnimHumanoid28 = (CHAnimHumanoid *)(m_pScene.createNode("HAnimHumanoid"));
HAnimHumanoid28->setDEF("hanim_humanoid");
HAnimHumanoid28->setName("humanoid");
HAnimHumanoid28->setLoa(0);
CMetadataSet* MetadataSet29 = (CMetadataSet *)(m_pScene.createNode("MetadataSet"));
MetadataSet29->setName("HAnimHumanoid.info");
MetadataSet29->setReference("https://www.web3d.org/documents/specifications/19774/V2.0/Architecture/ObjectInterfaces.html#Humanoid");
CMetadataString* MetadataString30 = (CMetadataString *)(m_pScene.createNode("MetadataString"));
MetadataString30->setName("authorEmail");
MetadataString30->setValue(new CString[1]{"beitler@graphics.cis.upenn.edu beitler@acm.org"}, 1);
MetadataSet29->setValue(*MetadataString30);

CMetadataString* MetadataString31 = (CMetadataString *)(m_pScene.createNode("MetadataString"));
MetadataString31->setName("authorName");
MetadataString31->setValue(new CString[1]{"Matthew T. Beitler"}, 1);
MetadataSet29->setValue(*MetadataString31);

CMetadataString* MetadataString32 = (CMetadataString *)(m_pScene.createNode("MetadataString"));
MetadataString32->setName("copyright");
MetadataString32->setValue(new CString[1]{"Copyright 1999 Matthew T. Beitler"}, 1);
MetadataSet29->setValue(*MetadataString32);

CMetadataString* MetadataString33 = (CMetadataString *)(m_pScene.createNode("MetadataString"));
MetadataString33->setName("creationDate");
MetadataString33->setValue(new CString[1]{"05/12/99"}, 1);
MetadataSet29->setValue(*MetadataString33);

CMetadataString* MetadataString34 = (CMetadataString *)(m_pScene.createNode("MetadataString"));
MetadataString34->setName("humanoidVersion");
MetadataString34->setValue(new CString[1]{"JointCenters 1.1 LOA0"}, 1);
MetadataSet29->setValue(*MetadataString34);

CMetadataString* MetadataString35 = (CMetadataString *)(m_pScene.createNode("MetadataString"));
MetadataString35->setName("usageRestrictions");
MetadataString35->setValue(new CString[1]{"PERMISSION TO FULLY USE THIS SCENE GRAPH IS GRANTED PROVIDED THIS COPYRIGHT INFORMATION AND DOCUMENTATION OF THE ORIGINAL AUTHOR IS INCLUDED. This humanoid scene graph is provided _as-is_ and without warranty of any kind express implied or otherwise including without limitation any warranty of merchantability or fitness for a particular purpose."}, 1);
MetadataSet29->setValue(*MetadataString35);

HAnimHumanoid28->setMetadata(*MetadataSet29);

CHAnimJoint* HAnimJoint36 = (CHAnimJoint *)(m_pScene.createNode("HAnimJoint"));
HAnimJoint36->setDEF("hanim_humanoid_root");
HAnimJoint36->setName("humanoid_root");
HAnimJoint36->setCenter(new float[3]{0,0.824,0.0277});
CHAnimJoint* HAnimJoint37 = (CHAnimJoint *)(m_pScene.createNode("HAnimJoint"));
HAnimJoint37->setDEF("hanim_sacroiliac");
HAnimJoint37->setName("sacroiliac");
HAnimJoint37->setCenter(new float[3]{0,0.9149,0.0016});
CHAnimSegment* HAnimSegment38 = (CHAnimSegment *)(m_pScene.createNode("HAnimSegment"));
HAnimSegment38->setDEF("hanim_pelvis");
HAnimSegment38->setName("pelvis");
CTransform* Transform39 = (CTransform *)(m_pScene.createNode("Transform"));
Transform39->setTranslation(new float[3]{0,0.9149,0.0016});
CShape* Shape40 = (CShape *)(m_pScene.createNode("Shape"));
Shape40->setDEF("DiamondShape");
CAppearance* Appearance41 = (CAppearance *)(m_pScene.createNode("Appearance"));
CMaterial* Material42 = (CMaterial *)(m_pScene.createNode("Material"));
Material42->setDiffuseColor(new float[3]{1,1,0});
Appearance41->setMaterial(*Material42);

Shape40->setAppearance(*Appearance41);

CIndexedFaceSet* IndexedFaceSet43 = (CIndexedFaceSet *)(m_pScene.createNode("IndexedFaceSet"));
IndexedFaceSet43->setCreaseAngle(0.5);
IndexedFaceSet43->setCoordIndex(new int[32]{0,1,2,-1,0,2,3,-1,0,3,4,-1,0,4,1,-1,5,2,1,-1,5,3,2,-1,5,4,3,-1,5,1,4,-1});
CCoordinate* Coordinate44 = (CCoordinate *)(m_pScene.createNode("Coordinate"));
Coordinate44->setPoint(new float[18]{0,0.01,0,-0.01,0,0,0,0,0.01,0.01,0,0,0,0,-0.01,0,-0.01,0});
IndexedFaceSet43->setCoord(*Coordinate44);

Shape40->setGeometry(IndexedFaceSet43);

Transform39->addChild(*Shape40);

HAnimSegment38->addChildren(*Transform39);

HAnimJoint37->addChildren(*HAnimSegment38);

HAnimJoint36->addChildren(*HAnimJoint37);

HAnimHumanoid28->setSkeleton(*HAnimJoint36);

CHAnimSite* HAnimSite45 = (CHAnimSite *)(m_pScene.createNode("HAnimSite"));
HAnimSite45->setDEF("hanim_site_view");
HAnimSite45->setName("site_view");
CViewpoint* Viewpoint46 = (CViewpoint *)(m_pScene.createNode("Viewpoint"));
Viewpoint46->setDEF("InclinedView");
Viewpoint46->setDescription("Inclined View");
Viewpoint46->setPosition(new float[3]{1.62,1.05,2.06});
Viewpoint46->setOrientation(new float[4]{-0.113,0.993,0.0347,0.671});
HAnimSite45->addChildren(*Viewpoint46);

CViewpoint* Viewpoint47 = (CViewpoint *)(m_pScene.createNode("Viewpoint"));
Viewpoint47->setDEF("FrontView");
Viewpoint47->setDescription("Front View");
Viewpoint47->setPosition(new float[3]{0,0.854,2.57665});
HAnimSite45->addChildren(*Viewpoint47);

CViewpoint* Viewpoint48 = (CViewpoint *)(m_pScene.createNode("Viewpoint"));
Viewpoint48->setDEF("SideView");
Viewpoint48->setDescription("Side View");
Viewpoint48->setPosition(new float[3]{2.5929,0.854,0});
Viewpoint48->setOrientation(new float[4]{0,1,0,1.57079});
HAnimSite45->addChildren(*Viewpoint48);

CViewpoint* Viewpoint49 = (CViewpoint *)(m_pScene.createNode("Viewpoint"));
Viewpoint49->setDEF("TopView");
Viewpoint49->setDescription("Top View");
Viewpoint49->setPosition(new float[3]{0,3.4495,0});
Viewpoint49->setOrientation(new float[4]{1,0,0,-1.57079});
HAnimSite45->addChildren(*Viewpoint49);

HAnimHumanoid28->addViewpoints(*HAnimSite45);

CHAnimSegment* HAnimSegment50 = (CHAnimSegment *)(m_pScene.createNode("HAnimSegment"));
HAnimSegment50->setUSE("hanim_pelvis");
HAnimHumanoid28->setSegments(*HAnimSegment50);

CHAnimJoint* HAnimJoint51 = (CHAnimJoint *)(m_pScene.createNode("HAnimJoint"));
HAnimJoint51->setUSE("hanim_humanoid_root");
HAnimHumanoid28->addJoints(*HAnimJoint51);

CHAnimJoint* HAnimJoint52 = (CHAnimJoint *)(m_pScene.createNode("HAnimJoint"));
HAnimJoint52->setUSE("hanim_sacroiliac");
HAnimHumanoid28->addJoints(*HAnimJoint52);

group->addChildren(*HAnimHumanoid28);

X3D0->setScene(*Scene24);

m_pScene.addRootNode(group);
X3D0->toXMLString();
}

File: src/main/python/net/coderextreme/data/DiamondManLOA0patched2.py
import x3dpsail as x3d
X3D0 = x3d.X3D()
X3D0.setProfile("Immersive")
X3D0.setVersion("4.0")
head1 = x3d.head()
component2 = x3d.component()
component2.setName("HAnim")
component2.setLevel(1)

head1.addComponent(component2)
meta3 = x3d.meta()
meta3.setName("title")
meta3.setContent("DiamondManLOA0patched2.x3d")

head1.addMeta(meta3)
meta4 = x3d.meta()
meta4.setName("description")
meta4.setContent("HAnim skeletal structure for Level of Action (LOA) zero, with one diamond at the base node for the structure. HumanoidRoot only, so this is the minimum legal HAnim humanoid.")

head1.addMeta(meta4)
meta5 = x3d.meta()
meta5.setName("creator")
meta5.setContent("Matthew T. Beitler")

head1.addMeta(meta5)
meta6 = x3d.meta()
meta6.setName("translator")
meta6.setContent("Joel S. Pawloski")

head1.addMeta(meta6)
meta7 = x3d.meta()
meta7.setName("created")
meta7.setContent("12 November 2001")

head1.addMeta(meta7)
meta8 = x3d.meta()
meta8.setName("modified")
meta8.setContent("2 November 2023")

head1.addMeta(meta8)
meta9 = x3d.meta()
meta9.setName("motto")
meta9.setContent("(a) \"Diamonds are a girl's best friend.\" (b) \"Gosh, it sure is chilly in here.\"")

head1.addMeta(meta9)
meta10 = x3d.meta()
meta10.setName("reference")
meta10.setContent("HAnim 2.0 specification, Appendix A: Nominal human body dimensions and levels of articulation (LOAs)")

head1.addMeta(meta10)
meta11 = x3d.meta()
meta11.setName("reference")
meta11.setContent("https://www.web3d.org/documents/specifications/19774-1/V2.0/HAnim/BodyDimensionsAndLOAs.html")

head1.addMeta(meta11)
meta12 = x3d.meta()
meta12.setName("reference")
meta12.setContent("https://www.web3d.org/documents/specifications/19774-1/V2.0/HAnim/BodyDimensionsAndLOAs.html#LevelOfArticulationZero")

head1.addMeta(meta12)
meta13 = x3d.meta()
meta13.setName("reference")
meta13.setContent("HAnim 1.1 specification, Appendix A: Suggested Body Dimensions and Levels of Articulation, Level of Articulation Zero")

head1.addMeta(meta13)
meta14 = x3d.meta()
meta14.setName("reference")
meta14.setContent("http://HAnim.org/Specifications/HAnim1.1/appendices.html#appendixa")

head1.addMeta(meta14)
meta15 = x3d.meta()
meta15.setName("reference")
meta15.setContent("http://HAnim.org/Specifications/HAnim1.1/JointCenters1_1_LOA0.wrl")

head1.addMeta(meta15)
meta16 = x3d.meta()
meta16.setName("reference")
meta16.setContent("http://HAnim.org/Specifications/HAnim1.1/JointCenters1_1_LOA0-diamond.wrl")

head1.addMeta(meta16)
meta17 = x3d.meta()
meta17.setName("reference")
meta17.setContent("http://ece.uwaterloo.ca/~HAnim")

head1.addMeta(meta17)
meta18 = x3d.meta()
meta18.setName("reference")
meta18.setContent("http://www.cis.upenn.edu/~badler/anthro/89-71.pdf")

head1.addMeta(meta18)
meta19 = x3d.meta()
meta19.setName("reference")
meta19.setContent("http://www.cis.upenn.edu/~badler/anthro/89-71.ps")

head1.addMeta(meta19)
meta20 = x3d.meta()
meta20.setName("reference")
meta20.setContent("http://www.cis.upenn.edu/~beitler")

head1.addMeta(meta20)
meta21 = x3d.meta()
meta21.setName("Image")
meta21.setContent("humanoid_landmark_locations.gif")

head1.addMeta(meta21)
meta22 = x3d.meta()
meta22.setName("Image")
meta22.setContent("http://HAnim.org/Specifications/HAnim1.1/humanoid_landmark_locations.gif")

head1.addMeta(meta22)
meta23 = x3d.meta()
meta23.setName("identifier")
meta23.setContent("https://www.web3d.org/x3d/content/examples/HumanoidAnimation/Templates/DiamondManLOA0.x3d")

head1.addMeta(meta23)
meta24 = x3d.meta()
meta24.setName("generator")
meta24.setContent("X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit")

head1.addMeta(meta24)
meta25 = x3d.meta()
meta25.setName("license")
meta25.setContent("../license.html")

head1.addMeta(meta25)

X3D0.setHead(head1)
Scene26 = x3d.Scene()
WorldInfo27 = x3d.WorldInfo()
WorldInfo27.setInfo(["HAnim 2.0 Default Joint Centers, Level Of Articulation (LOA) 0 -------------------------------------------------------- HANIM 1.1 (VRML 2.0) Author name: eMpTy (a.k.a. Matthew T. Beitler) HANIM 1.1 (VRML 2.0) Author email: beitler@graphics.cis.upenn.edu or beitler@acm.org HANIM 1.1 (VRML 2.0) Author homepage: http://www.cis.upenn.edu/~beitler HANIM 1.1 (VRML 2.0) Compliance Date: May 12, 1999 HANIM 1.1 Compliance Information: http://ece.uwaterloo.ca/~HAnim/ Construction Info (joint centers): The joint centers of this figure are based on the work of Norman Badler, director of the Center for Human Modeling and Simulation at the University of Pennsylvania. The original document which these joint centers are based on can be found at: http://www.cis.upenn.edu/~badler/anthro/89-71.ps, .pdf"])
WorldInfo27.setTitle("HANIM 2.0 Default Joint Centers, LOA0")

Scene26.addChildren(WorldInfo27)
NavigationInfo28 = x3d.NavigationInfo()
NavigationInfo28.setSpeed(1.5)

Scene26.addChildren(NavigationInfo28)
Viewpoint29 = x3d.Viewpoint()
Viewpoint29.setCenterOfRotation([0,1,0])
Viewpoint29.setDescription("Diamond Man, LOA 0")
Viewpoint29.setPosition([0,1,3])

Scene26.addChildren(Viewpoint29)
HAnimHumanoid30 = x3d.HAnimHumanoid()
HAnimHumanoid30.setName("humanoid")
HAnimHumanoid30.setDEF("hanim_humanoid")
HAnimHumanoid30.setLoa(0)
HAnimHumanoid30.setVersion("2.0")
#original HAnimHumanoid info='\"authorEmail=beitler@graphics.cis.upenn.edu beitler@acm.org\" \"authorName=Matthew T. Beitler\" \"copyright=Copyright 1999 Matthew T. Beitler\" \"creationDate=05/12/99\" \"humanoidVersion=JointCenters 1.1 LOA0\" \"usageRestrictions=PERMISSION TO FULLY USE THIS SCENE GRAPH IS GRANTED PROVIDED THIS COPYRIGHT INFORMATION AND DOCUMENTATION OF THE ORIGINAL AUTHOR IS INCLUDED. This humanoid scene graph is provided _as-is_ and without warranty of any kind express implied or otherwise including without limitation any warranty of merchantability or fitness for a particular purpose.\"'
MetadataSet31 = x3d.MetadataSet()
MetadataSet31.setName("HAnimHumanoid.info")
MetadataSet31.setReference("https://www.web3d.org/documents/specifications/19774/V2.0/Architecture/ObjectInterfaces.html#Humanoid")
MetadataString32 = x3d.MetadataString()
MetadataString32.setName("authorEmail")
MetadataString32.setValue(["beitler@graphics.cis.upenn.edu beitler@acm.org"])

MetadataSet31.setValue(MetadataString32)
MetadataString33 = x3d.MetadataString()
MetadataString33.setName("authorName")
MetadataString33.setValue(["Matthew T. Beitler"])

MetadataSet31.setValue(MetadataString33)
MetadataString34 = x3d.MetadataString()
MetadataString34.setName("copyright")
MetadataString34.setValue(["Copyright 1999 Matthew T. Beitler"])

MetadataSet31.setValue(MetadataString34)
MetadataString35 = x3d.MetadataString()
MetadataString35.setName("creationDate")
MetadataString35.setValue(["05/12/99"])

MetadataSet31.setValue(MetadataString35)
MetadataString36 = x3d.MetadataString()
MetadataString36.setName("humanoidVersion")
MetadataString36.setValue(["JointCenters 1.1 LOA0"])

MetadataSet31.setValue(MetadataString36)
MetadataString37 = x3d.MetadataString()
MetadataString37.setName("usageRestrictions")
MetadataString37.setValue(["PERMISSION TO FULLY USE THIS SCENE GRAPH IS GRANTED PROVIDED THIS COPYRIGHT INFORMATION AND DOCUMENTATION OF THE ORIGINAL AUTHOR IS INCLUDED. This humanoid scene graph is provided _as-is_ and without warranty of any kind express implied or otherwise including without limitation any warranty of merchantability or fitness for a particular purpose."])

MetadataSet31.setValue(MetadataString37)

HAnimHumanoid30.setMetadata(MetadataSet31)
HAnimJoint38 = x3d.HAnimJoint()
HAnimJoint38.setName("humanoid_root")
HAnimJoint38.setDEF("hanim_humanoid_root")
HAnimJoint38.setCenter([0,0.824,0.0277])
HAnimJoint39 = x3d.HAnimJoint()
HAnimJoint39.setName("sacroiliac")
HAnimJoint39.setDEF("hanim_sacroiliac")
HAnimJoint39.setCenter([0,0.9149,0.0016])
HAnimSegment40 = x3d.HAnimSegment()
HAnimSegment40.setName("pelvis")
HAnimSegment40.setDEF("hanim_pelvis")
Transform41 = x3d.Transform()
Transform41.setTranslation([0,0.9149,0.0016])
Shape42 = x3d.Shape()
Shape42.setDEF("DiamondShape")
IndexedFaceSet43 = x3d.IndexedFaceSet()
IndexedFaceSet43.setCoordIndex([0,1,2,-1,0,2,3,-1,0,3,4,-1,0,4,1,-1,5,2,1,-1,5,3,2,-1,5,4,3,-1,5,1,4,-1])
IndexedFaceSet43.setCreaseAngle(0.5)
Coordinate44 = x3d.Coordinate()
Coordinate44.setPoint([0,0.01,0,-0.01,0,0,0,0,0.01,0.01,0,0,0,0,-0.01,0,-0.01,0])

IndexedFaceSet43.setCoord(Coordinate44)

Shape42.setGeometry(IndexedFaceSet43)
Appearance45 = x3d.Appearance()
Material46 = x3d.Material()
Material46.setDiffuseColor([1,1,0])

Appearance45.setMaterial(Material46)

Shape42.setAppearance(Appearance45)

Transform41.addChild(Shape42)

HAnimSegment40.addChildren(Transform41)

HAnimJoint39.addChildren(HAnimSegment40)

HAnimJoint38.addChildren(HAnimJoint39)

HAnimHumanoid30.setSkeleton(HAnimJoint38)
HAnimSite47 = x3d.HAnimSite()
HAnimSite47.setName("site_view")
HAnimSite47.setDEF("hanim_site_view")
Viewpoint48 = x3d.Viewpoint()
Viewpoint48.setDEF("InclinedView")
Viewpoint48.setDescription("Inclined View")
Viewpoint48.setOrientation([-0.113,0.993,0.0347,0.671])
Viewpoint48.setPosition([1.62,1.05,2.06])

HAnimSite47.addChildren(Viewpoint48)
Viewpoint49 = x3d.Viewpoint()
Viewpoint49.setDEF("FrontView")
Viewpoint49.setDescription("Front View")
Viewpoint49.setPosition([0,0.854,2.57665])

HAnimSite47.addChildren(Viewpoint49)
Viewpoint50 = x3d.Viewpoint()
Viewpoint50.setDEF("SideView")
Viewpoint50.setDescription("Side View")
Viewpoint50.setOrientation([0,1,0,1.57079])
Viewpoint50.setPosition([2.5929,0.854,0])

HAnimSite47.addChildren(Viewpoint50)
Viewpoint51 = x3d.Viewpoint()
Viewpoint51.setDEF("TopView")
Viewpoint51.setDescription("Top View")
Viewpoint51.setOrientation([1,0,0,-1.57079])
Viewpoint51.setPosition([0,3.4495,0])

HAnimSite47.addChildren(Viewpoint51)

HAnimHumanoid30.setViewpoints(HAnimSite47)
HAnimJoint52 = x3d.HAnimJoint()
HAnimJoint52.setUSE("hanim_humanoid_root")

HAnimHumanoid30.addJoints(HAnimJoint52)
HAnimJoint53 = x3d.HAnimJoint()
HAnimJoint53.setUSE("hanim_sacroiliac")

HAnimHumanoid30.addJoints(HAnimJoint53)
HAnimSegment54 = x3d.HAnimSegment()
HAnimSegment54.setUSE("hanim_pelvis")

HAnimHumanoid30.setSegments(HAnimSegment54)

Scene26.addChildren(HAnimHumanoid30)

X3D0.setScene(Scene26)
X3D0.toFileX3D("../data/DiamondManLOA0patched2.new.python.x3d")
X3D0.toFileJSON("../data/DiamondManLOA0patched2.new.python.json")
